Middleware/SymfonyRouteResolver.php

```diff
@@ -132,4 +137,42 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
 
         return $response;
     }
+
+    /**
+     * Resolving the PageArguments (e.g. Typo3 page id) from the route.
+     * This is normally done by typo3 in a later middleware, but we want to have this information in our symfony request.
+     * Legacy URIs (?id=12345) are not supported for the time being (only routes).
+     */
+    protected function prepareForSymfonyRequest(ServerRequestInterface $request): ServerRequestInterface
+    {
+        $requestId = (string) ($request->getQueryParams()['id'] ?? '');
+
+        /** @var Site $site */
+        $site = $request->getAttribute('site', null);
+
+        if ($requestId || !($site instanceof  SiteInterface)) {
+            return $request;
+        }
+
+        $routeResult = $request->getAttribute('routing', null);
+        if (!($routeResult->getLanguage() instanceof SiteLanguage)) {
+            return $request;
+        }
+
+        try {
+            /** @var PageArguments $pageArguments */
+            $pageArguments = $site->getRouter()->matchRequest($request, $routeResult);
+        } catch (RouteNotFoundException $e) {
+            return $request;
+        }
+
+        $preparedRequest = $request->withAttribute('routing', $pageArguments);
+        // merge the PageArguments with the request query parameters
+        $queryParams = \array_replace_recursive(
+            $preparedRequest->getQueryParams(),
+            $pageArguments->getArguments()
+        );
```

Comment:
Why merge it with query params?

Reply:
I copied this part from the typo3 PageResolver middleware. I suspect it merges back any parameters resolved from the route.